src/vm/amd64/asmhelpers.S

```diff
@@ -24,23 +24,22 @@
 //        UINT64      flt1;
 //        UINT64      flt2;
 //        UINT64      flt3;
+//	#if defined(UNIX_AMD64_ABI)
+//	      UINT64      flt4;
+//	      UINT64      flt5;
+//	      UINT64      flt6;
+//	      UINT64      flt7;
+//	#endif
 //        UINT32      flags;
 //    } PROFILE_PLATFORM_SPECIFIC_DATA, *PPROFILE_PLATFORM_SPECIFIC_DATA;
 //
-.equ SIZEOF_PROFILE_PLATFORM_SPECIFIC_DATA, 0x8*11 + 0x4*2    // includes fudge to make FP_SPILL right
-.equ SIZEOF_OUTGOING_ARGUMENT_HOMES, 0x0
-.equ SIZEOF_FP_ARG_SPILL, 0x10*1
+.equ SIZEOF_PROFILE_PLATFORM_SPECIFIC_DATA, 0x8*15 + 0x4*2   // includes fudge to make FP_SPILL right
+.equ SIZEOF_FP_ARG_SPILL, 0x10*2
 
-// Need to be careful to keep the stack 16byte aligned here, since we are pushing 3
-// arguments that will align the stack and we just want to keep it aligned with our
-// SIZEOF_STACK_FRAME
+.equ OFFSETOF_FP_ARG_SPILL_0, SIZEOF_PROFILE_PLATFORM_SPECIFIC_DATA + 0x8
+.equ OFFSETOF_FP_ARG_SPILL_1, OFFSETOF_FP_ARG_SPILL_0 + 0x10
 
-.equ OFFSETOF_PLATFORM_SPECIFIC_DATA, SIZEOF_OUTGOING_ARGUMENT_HOMES
-
-// we'll just spill into the PROFILE_PLATFORM_SPECIFIC_DATA structure
-.equ OFFSETOF_FP_ARG_SPILL, SIZEOF_OUTGOING_ARGUMENT_HOMES + SIZEOF_PROFILE_PLATFORM_SPECIFIC_DATA
-
-.equ SIZEOF_STACK_FRAME, SIZEOF_OUTGOING_ARGUMENT_HOMES + SIZEOF_PROFILE_PLATFORM_SPECIFIC_DATA + SIZEOF_MAX_FP_ARG_SPILL
+.equ SIZEOF_STACK_FRAME, SIZEOF_PROFILE_PLATFORM_SPECIFIC_DATA + SIZEOF_MAX_FP_ARG_SPILL
```

Comment:
This gives you incorrect RSP alignment now. The RSP needs to be aligned to 16 bytes after the prolog of the function based on the calling convention. When function is entered, the RSP is aligned to 8 bytes and the RSP is moved by two pushes and the SIZEOF_STACK_FRAME (which is a multiple of 16 bytes - 0x8*15 + 0x4*2 +0x10*2), so the RSP is aligned to 8 bytes. 
You'll need to add 8 to the SIZEOF_STACK_FRAME.